client/src/app/domain/models/mediafiles/mediafile.constants.ts

```diff
@@ -19,16 +25,26 @@ export const LogoDisplayNames: { [place in LogoPlace]: string } = {
     pdf_footer_r: _(`PDF footer logo (right)`),
     pdf_ballot_paper: _(`PDF ballot paper logo`)
 };
+/**
+ * All different places where logos can be placed.
+ */
+export const LogoPlaces = Array.from(PossibleLogoPlaces.values());
```

Comment:
Generating an Array from the keys of `LogoDisplayNames` will just generate a string array (which I guess you could cast to be an `LogoPlace` array, but if you're going to have to process something anyway why not  generate it from the original data to begin with?)

`PossibleLogoPlaces` is cast as a `const ` so that I can use it to generate the Type. It will not be recognized as an array and therefore can't be used in places where the requirement is to have `(LogoPlace)[]`.
So I have to somehow generate an array from that => `LogoPlaces`

I could try to somehow generate `PossibleLogoPlaces` from the keys of `LogoDisplayNames`, but I believe that would mean that `LogoDisplayNames` isn't explicitly tied to the type any more, which also isn't very nice.

That is what led to this version,but  I can change it if it really bothers you.
What would you prefer instead?

Reply:
I would prefer only one place to gather the values and display names:
```
export const LogoPlaces = {
    projector_main: _(`Projector logo`),
    projector_header: _(`Projector header image`),
    web_header: _(`Web interface header logo`),
    pdf_header_l: _(`PDF header logo (left)`),
    pdf_header_r: _(`PDF header logo (right)`),
    pdf_footer_l: _(`PDF footer logo (left)`),
    pdf_footer_r: _(`PDF footer logo (right)`),
    pdf_ballot_paper: _(`PDF ballot paper logo`)
};
export type LogoPlace = keyof typeof LogoPlaces;
```
This way, there is no redundancy and one cannot e.g. add a new place without adding a display name for it. You can use `LogoPlace` just the way before and it should simplify the typing in using classes. What do you think?

Reply:
I can try it

Reply:
You could also leave the name `LogoDisplayNames` and add a `LogoPlaces = Object.keys(LogoDisplayNames)` if you don't want to write that everytime you access the keys. Also, what's the naming scheme in the client? Should `LogoPlaces` be called 'LOGO_PLACES` since it's a constant and not a type?